dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh

```diff
@@ -282,13 +282,39 @@ if ! oc get project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null; then
   if [ "${COMMAND}" == "cleanup" ] || [ "${COMMAND}" == "rollupdate" ]; then echo "**ERROR** project doesn't exist. Aborting"; exit 1; fi
   if [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then echo "**ERROR** project doesn't exist on OSIO. Aborting"; exit 1; fi
 
-  echo -n "Project does not exist...creating it..."
-  oc new-project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null
+  # OpenShift will not get project but project still exists for for a period after being deleted.
+  # The following will loop until it can create successfully.
+
+  WAIT_FOR_PROJECT_TO_DELETE=true
+  WAIT_FOR_PROJECT_TO_DELETE_MESSAGE="Waiting for project to be deleted fully(~15 seconds)..."
+
+  echo "Project \"${CHE_OPENSHIFT_PROJECT}\" does not exist...trying to creating it."
+  while $WAIT_FOR_PROJECT_TO_DELETE
+  do
+  { # try
+
+      oc new-project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null && \
```

Comment:
It can fail if another user has a project with the same name. So infinitive loop will be in this case. Please consider adding timeout for waiting.

Reply:
I don't think you can have two projects with the same name in OpenShift can you? This is one of the reason I created PR [https://github.com/eclipse/che/pull/8154](https://github.com/eclipse/che/pull/8154). I can add a timeout though anyways.

Reply:
@JamesDrummond You can't. So script expects that error is temporary (like a project is marked to remove and removing is in progress) and it will disappear when a project will be deleted. 
But error can be permanent - like another user has a project with the same name. Then infinitive loop will be here. 
P.S. Let me know if my explanation is not clear enough. I'll try to reword it.

Reply:
@sleshchenko In OpenShift I don't think it is possible to have two projects with the same name. Even seperate users can use a project name if it's already taken. I actually remember when I used osio this was even an issue. Took some screen captures below where james@example.com has eclipse-che project and developer user cannot create a project with same name.

![image](https://user-images.githubusercontent.com/20580888/34951140-e810d5b4-f9da-11e7-9671-a891eec76cfb.png)

![image](https://user-images.githubusercontent.com/20580888/34951112-d07b89c6-f9da-11e7-98df-bf755b8ab5b1.png)

![image](https://user-images.githubusercontent.com/20580888/34951105-c9ed47c0-f9da-11e7-9888-f15bf064c55b.png)

Reply:
everybody agrees that you can't have 2 projects with the same name, but what @sleshchenko meant that if that error will happen script will go into endless loop which is no OK and there is should be a timer or some kind of interruptor.

Reply:
@riuvshin Sorry didn't understand. Will add timer. Thanks.